### PR TITLE
sosreport for rhel-atomic

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
Image needs to be uploaded.

**Comment by mvollmer** [comment]
> Image needs to be uploaded.

Done.

File: test/check-sosreport [hunk 1]
```diff
@@ -35,11 +34,7 @@ class TestSOS(MachineCase):
         b.click('[data-target="#sos"]')
         b.wait_visible("#sos")
 
-        # HACK - https://bugzilla.redhat.com/show_bug.cgi?id=1278807
```
Comment (dperpeet) [inline test/check-sosreport]: Is this fixed in f23? the test timed out and there has been no final answer in the bug tracker so far.

File: test/check-sosreport [hunk 2]
```diff
@@ -35,11 +34,7 @@ class TestSOS(MachineCase):
...
-        # Sosreport waits 5 minutes for a hung ausearch, so we wait 10 minutes
-        # for sosreport.
-        with b.wait_timeout(600):
```
Comment (dperpeet) [inline test/check-sosreport]: In any case, isn't the default (I believe 60 seconds) very low here? I would think something like 2 minutes should always be allowed, especially on possibly slower test machines.

**Comment by dperpeet** [comment]
the updated spec file was merged into master, so this needs a rebase

File: pkg/sosreport/run-sosreport.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#! /bin/sh
```
Comment (dperpeet) [inline pkg/sosreport/run-sosreport.sh]: I think we should add a license header to new source files.

File: test/check-sosreport [hunk 4]
```diff
@@ -25,7 +25,6 @@ import subprocess
 import unittest
 
 @unittest.skipIf("debian" in os.environ.get("TEST_OS", ""), "No sosreport on Debian.")
```
Comment (dperpeet) [inline test/check-sosreport]: My debian 8 (jessie) has the package sosreport. Is that outdated or not compatible with what we do? Do we just need to build a new debian image?

File: pkg/sosreport/run-sosreport.sh (hunk 3 above)
Comment (stefwalter) [inline pkg/sosreport/run-sosreport.sh]: This is a script that traverses the network each time and doesn't get minified. I don't think it needs a license.

File: pkg/sosreport/run-sosreport.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+if grep -q 'CPE_NAME=.*atomic-host' /etc/os-release; then
```
Comment (stefwalter) [inline pkg/sosreport/run-sosreport.sh]: Can we just take this branch if the atomic executable is available?

File: test/check-sosreport (hunk 1 above)
Comment (stefwalter) [inline test/check-sosreport]: Because of the pty above, this is no longer necessary.

**Comment by stefwalter** [comment]
Needs a rebase onto master.

File: pkg/sosreport/run-sosreport.sh (hunk 5 above)
Comment (mvollmer) [inline pkg/sosreport/run-sosreport.sh]: Fedora also the atomic command, but we want to run sosreport directly there.  So we could check `atomic && !sosreport`.  But then we might use the wrong image... so I thought it is best to just detect rhel atomic directly....  What do you think?

File: test/check-sosreport (hunk 2 above)
Comment (mvollmer) [inline test/check-sosreport]: I think so too.

File: pkg/sosreport/run-sosreport.sh (hunk 5 above)
Comment (stefwalter) [inline pkg/sosreport/run-sosreport.sh]: Good point about the container. No way to predict that. However the CPE_NAME thingy above doesn't help with using the right image.

File: test/check-sosreport (hunk 4 above)
Comment (mvollmer) [inline test/check-sosreport]: You are right, I must have missed that somehow at the time.  I'll make a separate PR for this.

**Pull request closed without merging**